pkg/push/api/api.go

```diff
@@ -236,20 +241,149 @@ func (s *PushService) CreatePush(
+	return &pushproto.CreatePushResponse{
+		Push: push.Push,
+	}, nil
 }
 
-func (s *PushService) validateCreatePushRequest(req *pushproto.CreatePushRequest, localizer locale.Localizer) error {
-	if req.Command == nil {
-		dt, err := statusNoCommand.WithDetails(&errdetails.LocalizedMessage{
+// createPushNoCommand implement logic without command
+func (s *PushService) createPushNoCommand(
+	ctx context.Context,
+	req *pushproto.CreatePushRequest,
+	localizer locale.Localizer,
+	editor *eventproto.Editor,
+) (*pushproto.CreatePushResponse, error) {
+	if err := s.validateCreatePushRequestV2(req, localizer); err != nil {
+		return nil, err
+	}
+	push, err := domain.NewPush(
+		req.Name,
+		string(req.FcmServiceAccount),
+		req.Tags,
+	)
+	if err != nil {
+		s.logger.Error(
+			"Failed to create a new push",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("environmentNamespace", req.EnvironmentNamespace),
+				zap.Strings("tags", req.Tags),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
 			Locale:  localizer.GetLocale(),
-			Message: localizer.MustLocalizeWithTemplate(locale.RequiredFieldTemplate, "command"),
+			Message: localizer.MustLocalize(locale.InternalServerError),
 		})
 		if err != nil {
-			return statusInternal.Err()
+			return nil, statusInternal.Err()
 		}
-		return dt.Err()
+		return nil, dt.Err()
+	}
+	pushes, err := s.listAllPushes(ctx, req.EnvironmentNamespace, localizer)
+	if err != nil {
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+	if err := s.checkFCMServiceAccount(ctx, pushes, req.FcmServiceAccount, localizer); err != nil {
+		return nil, err
+	}
+	err = s.containsTags(pushes, req.Tags, localizer)
+	if err != nil {
+		if status.Code(err) == codes.AlreadyExists {
+			dt, err := statusTagAlreadyExists.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalize(locale.AlreadyExistsError),
+			})
+			if err != nil {
+				return nil, statusInternal.Err()
+			}
+			return nil, dt.Err()
+		}
+		s.logger.Error(
+			"Failed to validate tag existence",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+				zap.String("environmentNamespace", req.EnvironmentNamespace),
+				zap.Strings("tags", req.Tags),
+			)...,
+		)
+		dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+			Locale:  localizer.GetLocale(),
+			Message: localizer.MustLocalize(locale.InternalServerError),
+		})
+		if err != nil {
+			return nil, statusInternal.Err()
+		}
+		return nil, dt.Err()
+	}
+
+	var event *eventproto.Event
+	pushStorage := v2ps.NewPushStorage(s.mysqlClient)
+	err = pushStorage.CreatePush(ctx, push, req.EnvironmentNamespace)
```

Comment:
For creating and updating operations, please rum them using a transaction.

Reference:
https://github.com/bucketeer-io/bucketeer/blob/15a8681012601abfbee96be1a6b849420cbf525d/pkg/feature/api/feature.go#L586-L592

Reply:
Done, I have moved to transaction